src/iden3comm/handlers/payment.ts

```diff
@@ -47,15 +55,125 @@ export function createPaymentRequest(
+ * @param {DID} sender - sender did
+ * @param {DID} receiver - receiver did
+ * @param {Signer} signer - receiver did
+ * @param {string} agent - agent URL
+ * @param opts - payment options
+ * @returns {Promise<PaymentRequestMessage>}
+ */
+export async function createPaymentRailsV1(
+  sender: DID,
+  receiver: DID,
+  agent: string,
+  signer: Signer,
+  opts: {
+    payments: [
+      {
+        credentials: {
+          type: string;
+          context: string;
+        }[];
+        description?: string;
+        chains: {
+          nonce: bigint;
+          amount: bigint;
+          currency: SupportedCurrencies;
+          chainId: string;
+          recipient: string;
+          verifyingContract: string;
+          expirationDate?: Date;
+        }[];
+      }
+    ];
+  }
+): Promise<PaymentRequestMessage> {
+  const payments: PaymentRequestInfo[] = [];
+  for (let i = 0; i < opts.payments.length; i++) {
+    const { credentials, description } = opts.payments[i];
+    const dataArr: Iden3PaymentRailsRequestV1[] = [];
+    for (let j = 0; j < opts.payments[i].chains.length; j++) {
+      const { nonce, amount, currency, chainId, recipient, verifyingContract, expirationDate } =
+        opts.payments[i].chains[j];
+
+      if (recipient !== (await signer.getAddress())) {
+        throw new Error('recipient is not the signer');
+      }
+      const typeUrl = 'https://schema.iden3.io/core/json/Iden3PaymentRailsRequestV1.json';
+      const typesFetchResult = await fetch(typeUrl);
+      const types = await typesFetchResult.json();
+      delete types.EIP712Domain;
+      const paymentData = {
+        recipient,
+        amount,
+        expirationDate: expirationDate?.getTime() ?? 0,
+        nonce,
+        metadata: '0x'
+      };
+
+      const domain = {
+        name: 'MCPayment',
+        version: '1.0.0',
+        chainId,
+        verifyingContract
+      };
+      const signature = await signer.signTypedData(domain, types, paymentData);
+      dataArr.push({
+        type: PaymentRequestDataType.Iden3PaymentRailsRequestV1,
+        '@context': [
+          'https://schema.iden3.io/core/jsonld/payment.jsonld#Iden3PaymentRailsRequestV1',
+          'https://w3id.org/security/suites/eip712sig-2021/v1'
+        ],
+        recipient,
+        amount: amount.toString(),
+        currency,
+        expirationDate: expirationDate?.toISOString() ?? '',
+        nonce: nonce.toString(),
+        metadata: '0x',
+        proof: [
+          {
+            type: SupportedPaymentProofType.EthereumEip712Signature2021,
+            proofPurpose: 'assertionMethod',
+            proofValue: signature,
+            verificationMethod: `did:pkh:eip155:${chainId}:${recipient}#blockchainAccountId`,
+            created: new Date().toISOString(),
+            eip712: {
+              types: typeUrl,
+              primaryType: 'Iden3PaymentRailsRequestV1',
+              domain: {
+                ...domain,
+                salt: ''
```

Comment:
why salt is empty?

Reply:
Because now we don't use it in domain separator (name, version, chain id and contract address is enough). 
I believe we can remove this field since it's not even in the contract domain type: 
`keccak256("EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)");`

Reply:
removed